### PR TITLE
Add scatter for CUDA support

### PR DESCRIPTION
**Pull request by yuehhua**

### CODE REVIEW

**Comment by yuehhua** [comment]
I faced some issue while running into these error:

```julia
julia> using NNlibCUDA
[ Info: Precompiling NNlibCUDA [a00861dc-f156-4864-bf3c-e6376f28a68d]
ERROR: LoadError: ArgumentError: Package CompilerSupportLibraries_jll does not have JLLWrappers in its dependencies:
- If you have CompilerSupportLibraries_jll checked out for development and have
  added JLLWrappers as a dependency but haven't updated your primary
  environment's manifest file, try `Pkg.resolve()`.
- Otherwise you may need to report an issue with CompilerSupportLibraries_jll
```

It doesn't resolve when I execute `]resolve`

Any idea?

c.c.: @CarloLucibello

File: lib/NNlibCUDA/test/scatter.jl [hunk 1]
```diff
@@ -0,0 +1,147 @@
+dsts = Dict(
+    0 => cu([3, 4, 5, 6, 7]),
+    1 => cu([3 3 4 4 5;
+             5 5 6 6 7]),
+)
+srcs = Dict(
```
Comment (DhairyaLGandhi) [inline lib/NNlibCUDA/test/scatter.jl:6]: Could we represent them like for loops with multiple iterators?

**Comment by CarloLucibello** [comment]
Try with 
```julia
] add CUDA#master
```

**Comment by CarloLucibello** [comment]
Also need Julia 1.6

**Comment by yuehhua** [comment]
Sorry, using v1.6 and CUDA#master still cannot resolve the error.

**Comment by CarloLucibello** [comment]
@yuehhua I can't reproduce your problem. For me moving to the lib/NNlibCUDA directory, opening julia 1.6, `] activate .`, ] add CUDA#master`, and `using NNlibbCUA` works fine

**Comment by yuehhua** [comment]
It seems that there are some pre-compiling errors in my side.

```julia
[yuehhua@yuehhua-GL65 NNlib.jl]$ cd lib/NNlibCUDA/
[yuehhua@yuehhua-GL65 NNlibCUDA]$ julia-1.6.0 --proj
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.0-rc3 (2021-03-16)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(NNlibCUDA) pkg> up
    Updating registry at `~/.julia/registries/General`
    Updating git-repo `https://github.com/JuliaGPU/CUDA.jl.git`
    Updating `/media/yuehhua/Workbench/workspace/NNlib.jl/lib/NNlibCUDA/Project.toml`
  [052768ef] ~ CUDA v2.6.0 `https://github.com/JuliaGPU/CUDA.jl.git#master` ⇒ v2.6.0 `https://github.com/JuliaGPU/CUDA.jl.git#master`
    Updating `/media/yuehhua/Workbench/workspace/NNlib.jl/lib/NNlibCUDA/Manifest.toml`
  [052768ef] ~ CUDA v2.6.0 `https://github.com/JuliaGPU/CUDA.jl.git#master` ⇒ v2.6.0 `https://github.com/JuliaGPU/CUDA.jl.git#master`
Precompiling project...
  Progress [========================================>]  18/18
  ✗ MbedTLS_jll
  ✗ Zlib_jll
  ✗ CompilerSupportLibraries_jll
  ✗ CUDA
  ✗ NNlibCUDA
13 dependencies successfully precompiled in 18 seconds (10 already precompiled)
5 dependencies errored
```

**Comment by CarloLucibello** [comment]
I can reproduce now. Seems to be due to some problem in CUDA.jl, I filed an issue https://github.com/JuliaGPU/CUDA.jl/issues/787

**Comment by CarloLucibello** [comment]
So not a bug, you just need
```

**Comment by CarloLucibello** [comment]
based on the buildkite run, it seems that some of the gpu tests fail because they fallback to NNlib's scatter implementation

**Comment by yuehhua** [comment]
Current PR has been moved to FluxML/NNlibCUDA.jl#1, so close this one.

**Pull request closed without merging**